beekeeper-cleanup/src/test/java/com/expediagroup/beekeeper/cleanup/hive/HiveMetadataCleanerTest.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright (C) 2019-2020 Expedia, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.expediagroup.beekeeper.cleanup.hive;
+
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import com.expediagroup.beekeeper.cleanup.monitoring.DeletedMetadataReporter;
+import com.expediagroup.beekeeper.core.config.MetadataType;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+
+@ExtendWith(MockitoExtension.class)
+public class HiveMetadataCleanerTest {
+
+  private @Mock HousekeepingMetadata housekeepingMetadata;
```

Comment:
Not sure about this suggestion but I don't think there's any need to use a mock `HousekeepingMetadata` instance here when we can just use an actual one. You can remove the verify calls because the `verify(hiveClient).dropTable(DATABASE, TABLE_NAME)` indirectly verifies that the right methods are called.  You can also spy the actual object if you want and the verify the method calls.

I've always been under the impression to only use mocks when it is too difficult to create the object. @patduin care to share some of your wisdom with us? :-)